### DISCUSSION TITLE
Resource polymorphism

### DISCUSSION THREAD

**Discussion by tomsej** [dagster-io/dagster#21093] [Q&A] [answered]
Hi, I have a 2 dbt resources. One running on **Snowflake** and the second one in **BigQuery**. Both are living in different project folders, e.g.:

```
resource_def = {
    "local": {
        ....
        "dbt_sf": DbtCliResource(project_dir=DBT_PROJECT_DIR_SF, target="local"),
        "dbt_bq": DbtCliResource(project_dir=DBT_PROJECT_DIR_BQ, target="local"),
    },
}
```

Then I have an asset factory (it is used for running dbt operations), e.g.:

```
assets_to_make = ["a", "b", "c"]

def make_assets(asset_to_make):
    @asset(
        name=asset_to_make
    )
    def asset_template(dbt: DbtCliResource):
         dbt.cli(...)

    return asset_template
```

Is there a recommended way to pass resources to the ﻿`make_assets` function? It seems that the argument name of the function must match a key in the resource definition. While I could create a custom resource that includes both ﻿`dbt_sf` and ﻿`dbt_bq`, I am wondering if there is a more optimal solution available.

Thank you

**Comment by garethbrickman** [comment, marked as answer]
I think you're on the correct approach. There may be some more relevant info/guidance [in this related post](https://github.com/dagster-io/dagster/discussions/19079#discussioncomment-8055520)
